src/components/colorscale/color_attributes.js

```diff
@@ -64,14 +69,16 @@ module.exports = function makeColorScaleAttributes(context) {
         autocolorscale: extendDeep({}, colorScaleAttributes.autocolorscale, {
```

Comment:
Is it still true that both or neither of cmin,cmax must be set? My experience with `scatter3d` showed that omitting one of them did the reasonable thing: it behaved as if the omitted cm... value were set to the lowest (or highest) value of the `color` numeric array.